### PR TITLE
add json feed

### LINKED ISSUE #662: We need to improve the RSS feed
Gents,

I thought to give the sponsorlist atom feed a chance today...

http://changelog.qgis.org/en/qgis/sponsors/rss/

http://changelog.qgis.org/en/qgis/sponsors/atom/

Some things I think we have to tackle:

* Images

Should we pull all images from the changelog server
http://changelog.qgis.org

OR should we do some downloading of a zip with all stuff and let it
unzip by some cronjob?

An example:

http://changelog.qgis.org/media/images/projects/5f50b148aecade6339b100f1ad30897b81ca9a09.jpg

* Http vs HttpS

Now that the website is 100% https, all links to images etc should also
go to httpS...
Which makes pointing to a http://changelog.qgis.org only possible if we
create a reverseproxy for it on qgis2 ?

* Sorting in list(s)

Currently both atom and rss entry are not easily suited to be used in
some javascript:

ATOM entry:

 <entry>
   <title>QGIS user group Switzerland</title>
   <link
href="http://changelog.kartoza.com/en/qgis/sponsorshipperiod/dwlwhd/"
rel="alternate"/>
   <id>http://changelog.kartoza.com/en/qgis/sponsorshipperiod/dwlwhd/</id>
   <summary type="html">&lt;div&gt;&lt;img
src="/media/images/projects/945e5e1463ca738057fd9bae0a427519ccbfef0d.png"
width="300px"&gt;&lt;/div&gt;&lt;p&gt;&lt;span&gt;Sponsorship level:
Gold : 9000 EUR&lt;/span&gt;&lt;br/&gt;&lt;span&gt;Sponsorship period:
21 June 201>
 </entry>

RSS entry:

   <item>
     <title>QGIS user group Switzerland</title>

<link>http://changelog.kartoza.com/en/qgis/sponsorshipperiod/dwlwhd/</link>
     <description>&lt;div&gt;&lt;img
src="/media/images/projects/945e5e1463ca738057fd9bae0a427519ccbfef0d.png"
width="300px"&gt;&lt;/div&gt;&lt;p&gt;&lt;span&gt;Sponsorship level:
Gold : 9000 EUR&lt;/span&gt;&lt;br/&gt;&lt;span&gt;Sponsorship period:
21 June 2017 - 21>

<guid>http://changelog.kartoza.com/en/qgis/sponsorshipperiod/dwlwhd/</guid>
   </item>

To know if it is a gold or silver or bronze sponsor I would need to
regexp in the description. Which is doable by the way...

But to order the sponsors. I think they are ordered by end-time of
sponsorship I have to pull out the Sponsorship period and try to parse
both begin and endtime etc etc

Can we maybe come up with a javascript friendlier format?
FYI this
https://github.com/qgis/QGIS-Website/blob/master/themes/qgis-theme/languageswitch.html
Is the already becoming pretty complicated script to handle the feeds we
have now...

Happy to do skype call if easier.

Regards,

Richard Duivenvoorde

CC @rduivenvoorde

### PR DESCRIPTION
**Pull request by ann26**
fix #662 
- [x] Added JSON feed for sponsors.

JSON feed view:
screenshot from 2018-03-28 12-28-27

JSON feed icon on the sponsors list page:
The icon is similar with rss feed icon but it has different color on hover.
screenshot from 2018-03-28 12-27-54

- [x] Also made the navbar thinner.
Before fix:
screenshot from 2018-03-28 11-11-29

After fix:
screenshot from 2018-03-28 11-11-46

### CODE REVIEW

File: django_project/changes/feeds/json_rss_feed.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+            root_elements['categories'] = []
+        for cat in self.feed['categories']:
+            root_elements['categories'] += [cat, ]
+        if self.feed['feed_copyright'] is not None:
+            root_elements['copyright'] = self.feed['feed_copyright']
+        root_elements['lastBuildDate'] = rfc2822_date(self.latest_post_date())
+        if self.feed['ttl'] is not None:
+            root_elements['ttl'] = self.feed['ttl']
+
+        return root_elements
```
Comment (ismailsunni) [inline django_project/changes/feeds/json_rss_feed.py]: __[Flake8]__

[W292] no newline at end of file

File: django_project/changes/feeds/json_rss_feed.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        item_elements = {}
+
+        item_elements['title'] = item['title']
+        item_elements['link'] = item['link']
+        if item['description'] is not None:
+            item_elements['description'] = item['description']
+
+        if item['author_name'] and item['author_email']:
+            item_elements['author'] = "%s (%s)" % (
+            item['author_name'], item['author_email'])
```
Comment (ismailsunni) [inline django_project/changes/feeds/json_rss_feed.py]: __[Flake8]__

[E122] continuation line missing indentation or outdented

**Comment by timlinux** [comment]
Can you make the tests pass?

**Comment by ismailsunni** [comment]
Please add the unit test. See that the codecov is decreasing.

**Comment by ann26** [comment]
hi @timlinux and @ismailsunni , tests are passed :D

**Comment by timlinux** [comment]
I'm going to merge this and deploy...

**Pull request merged**